clang/lib/Sema/SemaChecking.cpp

```diff
@@ -2126,14 +2219,16 @@ static ExprResult PointerAuthSignGenericData(Sema &S, CallExpr *Call) {
 }
 
 static ExprResult PointerAuthSignOrAuth(Sema &S, CallExpr *Call,
-                                        PointerAuthOpKind OpKind) {
+                                        PointerAuthOpKind OpKind,
+                                        bool RequireConstant) {
   if (S.checkArgCount(Call, 3))
     return ExprError();
   if (checkPointerAuthEnabled(S, Call))
     return ExprError();
-  if (checkPointerAuthValue(S, Call->getArgs()[0], OpKind) ||
+  if (checkPointerAuthValue(S, Call->getArgs()[0], OpKind, RequireConstant) ||
```

Comment:
Can we check code like `Call->getArgs()[N]` to `Call->getArg(N)` while here? Certainly not everywhere, but on changed lines?

Reply:
This one is actually interesting: we can't do that as-is, because the helper functions take an `Expr *&`, to do type conversion and placeholder expansion in-place;  see e.g., `convertArgumentToType`.
Another existing user makes the modification explicit using a local and `setArg`, but we have a lot of these calls in the checking code here.

We can rename the helper to something explicit though, perhaps `checkAndConvertPointerAuthValue`, and pass the call expr and arg index separately (same for key etc.)

Reply:
Yes, `checkAndConvertPointerAuthValue` looks better to me as otherwise it is not pretty clear there are side effects.

Reply:
Let's do that in a separate PR and adopt it for the other builtins there